### ISSUE TITLE
HEAVILY reduced spawn rates.

### ISSUE THREAD

**Issue by MsVoxxie** [MarkusBordihn/adaptive_performance_tweaks#5]
**Describe the issue**

Hey there. I've been working on a new modpack for 1.16.5 and found your mod, Love what it does and how it works, however I've noticed a very heavy reduction in mob spawns, to the point that only 3-4 spawn near one player and the rest of the server gets zero spawns. I've increased the limits heavily for testing purposes, even increased the view distance settings yet nothing seemed to work, even disabling the spawn limiting modules of your mod.

Removing the mod entirely from the server returned mob spawns to normal.

**To Reproduce**
Play the game

**Expected behavior**
Normal / Little impact on mob spawning.

**Screenshots**
No screenshots.

**Debugging**
I've ran debugging and for my self in creative it would constantly block spawns, even with their limits much higher without any other players online.

**Additional context**
I've provided a full list of mods in my pack [here](https://share.voxxie.me/Viewer/../i/2021-04-22_14.28.35.html)

I certainly hope to find the cause or a solution because I quite enjoy your mod.

**Comment by MarkusBordihn** [comment]
Thanks for the report, could you please add some of the "debug" lines to see the main reason why the spawns are blocked like:
```
[18Apr.2021 14:35:39.330] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT] ? Enable debug output, please check debug.log for the full output.
[18Apr.2021 14:35:39.331] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT] Use '/aptweaks debug false' to disable debug output!
[18Apr.2021 14:35:39.341] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for SquidEntity['Squid'/2440, l='ServerLevel[Test World]', x=-62.50, y=61.00, z=-20.50] in minecraft:overworld.
[18Apr.2021 14:35:39.341] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/2443, l='ServerLevel[Test World]', x=-144.50, y=29.00, z=3.50] in minecraft:overworld.
[18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6311, l='ServerLevel[Test World]', x=-44.50, y=23.00, z=132.50] in minecraft:overworld.
[18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6312, l='ServerLevel[Test World]', x=-44.50, y=23.00, z=133.50] in minecraft:overworld.
[18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6313, l='ServerLevel[Test World]', x=34.50, y=10.00, z=95.50] in minecraft:overworld.
[18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6314, l='ServerLevel[Test World]', x=33.50, y=10.00, z=87.50] in minecraft:overworld.
[18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:skeleton (2 >= 2 * 1.0) in minecraft:overworld
[18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for ZombieEntity['Zombie'/6316, l='ServerLevel[Test World]', x=-123.50, y=32.00, z=20.50] in minecraft:overworld.
[18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6318, l='ServerLevel[Test World]', x=-77.50, y=19.00, z=162.50] in minecraft:overworld.
[18Apr.2021 14:35:56.293] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6321, l='ServerLevel[Test World]', x=-8.50, y=41.00, z=159.50] in minecraft:overworld.
```

If you change the config you need to restart the game. I will change this in future updates but at the moment the config is cached on start for performance reasons.

The modlist looks fine, I'm using most of the mods as well, but the debug lines would be helpful to understand the reason. Thanks.

**Comment by MsVoxxie** [comment]
> Thanks for the report, could you please add some of the "debug" lines to see the main reason why the spawns are blocked like:
> 
> ```
> [18Apr.2021 14:35:39.330] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT] ? Enable debug output, please check debug.log for the full output.
> [18Apr.2021 14:35:39.331] [Render thread/INFO] [net.minecraft.client.gui.NewChatGui/]: [CHAT] Use '/aptweaks debug false' to disable debug output!
> [18Apr.2021 14:35:39.341] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for SquidEntity['Squid'/2440, l='ServerLevel[Test World]', x=-62.50, y=61.00, z=-20.50] in minecraft:overworld.
> [18Apr.2021 14:35:39.341] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/2443, l='ServerLevel[Test World]', x=-144.50, y=29.00, z=3.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6311, l='ServerLevel[Test World]', x=-44.50, y=23.00, z=132.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6312, l='ServerLevel[Test World]', x=-44.50, y=23.00, z=133.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6313, l='ServerLevel[Test World]', x=34.50, y=10.00, z=95.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6314, l='ServerLevel[Test World]', x=33.50, y=10.00, z=87.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.243] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:skeleton (2 >= 2 * 1.0) in minecraft:overworld
> [18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for ZombieEntity['Zombie'/6316, l='ServerLevel[Test World]', x=-123.50, y=32.00, z=20.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
> [18Apr.2021 14:35:56.292] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6318, l='ServerLevel[Test World]', x=-77.50, y=19.00, z=162.50] in minecraft:overworld.
> [18Apr.2021 14:35:56.293] [Server thread/DEBUG] [APTweaks/]: [View Area Visibility] Blocked spawn event for BatEntity['Bat'/6321, l='ServerLevel[Test World]', x=-8.50, y=41.00, z=159.50] in minecraft:overworld.
> ```
> 
> If you change the config you need to restart the game. I will change this in future updates but at the moment the config is cached on start for performance reasons.
> 
> The modlist looks fine, I'm using most of the mods as well, but the debug lines would be helpful to understand the reason. Thanks.

Of course, [Here](https://share.voxxie.me/i/2021-04-22_17.30.01.txt) is the full debug log.

And Each configuration edit I have done I restarted the entire server, considering that should be the configuration that is affecting the spawns.

**Comment by MarkusBordihn** [comment]
Thanks a lot for the debug log. APTWeaks assumes the following settings according the log:
```
[22Apr2021 00:30:01.722] [Server thread/INFO] [APTweaks/]: Max number of Items allowed per world: 1000 / per type: 100
[22Apr2021 00:30:01.722] [Server thread/INFO] [APTweaks/]: Pre-calculate entity spawns rates for Players and World
[22Apr2021 00:30:01.727] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for Alex's Mobs and 10 mobs with maxPerPlayer:2 and maxPerWorld:8 ...
[22Apr2021 00:30:01.727] [Server thread/INFO] [APTweaks/]: ? Enable neutral mobs spawn rate control for Alex's Mobs and 19 mobs with maxPerPlayer:2 and maxPerWorld:8 ...
[22Apr2021 00:30:01.727] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Alex's Mobs and 17 mobs with maxPerPlayer:1 and maxPerWorld:10 ...
[22Apr2021 00:30:01.729] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for Aquaculture 2 and 40 mobs with maxPerPlayer:1 and maxPerWorld:8 ...
[22Apr2021 00:30:01.736] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Mekanism Additions and 5 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.737] [Server thread/INFO] [APTweaks/]: ? Enable special mobs spawn rate control for Minecraft and 6 mobs with: {minecraft:the_nether:minecraft:piglin_brute=20, minecraft:the_nether:minecraft:piglin=20, minecraft:the_nether:zombified_piglin=20, minecraft:the_nether:wither_skeleton=10, minecraft:the_end:minecraft:enderman=40, minecraft:the_nether:minecraft:skeleton=20}
[22Apr2021 00:30:01.738] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for Minecraft and 1 mobs with maxPerPlayer:1 and maxPerWorld:6 ...
[22Apr2021 00:30:01.738] [Server thread/INFO] [APTweaks/]: ? Enable neutral mobs spawn rate control for Minecraft and 24 mobs with maxPerPlayer:4 and maxPerWorld:20 ...
[22Apr2021 00:30:01.738] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Minecraft and 36 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.738] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for Minecraft (water) and 4 mobs with maxPerPlayer:1 and maxPerWorld:10 ...
[22Apr2021 00:30:01.738] [Server thread/INFO] [APTweaks/]: ? Enable neutral mobs spawn rate control for Minecraft (water) and 2 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.741] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for Quark and 3 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.741] [Server thread/INFO] [APTweaks/]: ? Enable neutral mobs spawn rate control for Quark and 1 mobs with maxPerPlayer:1 and maxPerWorld:10 ...
[22Apr2021 00:30:01.741] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Quark and 3 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.744] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Savage and Ravage and 4 mobs with maxPerPlayer:4 and maxPerWorld:10 ...
[22Apr2021 00:30:01.748] [Server thread/INFO] [APTweaks/]: ? Enable passive mobs spawn rate control for The Abyss and 10 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
[22Apr2021 00:30:01.748] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for The Abyss and 30 mobs with maxPerPlayer:4 and maxPerWorld:10 ...
[22Apr2021 00:30:01.749] [Server thread/INFO] [APTweaks/]: ? Enable hostile mobs spawn rate control for Tinkers Construct and 1 mobs with maxPerPlayer:4 and maxPerWorld:16 ...
[22Apr2021 00:30:01.749] [Server thread/INFO] [APTweaks/]: Added 210 player spawn rules, 210 world spawn rules and 6 special spawn rules.
```

Example:
...Enable hostile mobs spawn rate control for Minecraft and 36 mobs with maxPerPlayer:2 and maxPerWorld:10 ...
means max 2. hostile Minecraft mobs per Player and max. 10 mobs per world.

This is matching with the log entries like:
```
[22Apr2021 00:36:43.904] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:atlantic_herring (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.904] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:red_grouper (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.904] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:atlantic_herring (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.904] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.904] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:dolphin (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.905] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:atlantic_herring (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.905] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:atlantic_halibut (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.905] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:zombie (3 >= 2 * 1.5) in minecraft:overworld
[22Apr2021 00:36:43.906] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:skeleton (3 >= 2 * 1.5) in minecraft:overworld
[22Apr2021 00:36:43.906] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:spider (3 >= 2 * 1.5) in minecraft:overworld
[22Apr2021 00:36:43.908] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:bat (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.908] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.908] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:dolphin (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.909] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:bat (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.909] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:atlantic_herring (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.909] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:jellyfish (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.910] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.910] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:dolphin (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.910] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.910] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for ars_nouveau:wilden_stalker (15 >= 10 * 1.5) in minecraft:overworld
[22Apr2021 00:36:43.910] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:tuna (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:cod (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for aquaculture:red_grouper (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:squid (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:bat (1 >= 1 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for alexsmobs:cockroach (2 >= 2 * 1.0) in minecraft:overworld
[22Apr2021 00:36:43.911] [Server thread/DEBUG] [APTweaks/]: [Player limit] Blocked spawn event for minecraft:bat (1 >= 1 * 1.0) in minecraft:overworld
```

This means it seems that your configuration changes are not regonized by APTweaks.
Could you please send me your modified **adaptive_performance_tweaks-common.toml** to check if the same config file works fine on my side? Thanks.

**Comment by MsVoxxie** [comment]
How odd, I've tried so many things with the config so maybe I messed up somewhere some how, [Heres my config](https://share.voxxie.me/i/2021-04-22_19.55.00.toml).

I appreciate you taking the time to look into this :)

**Comment by MarkusBordihn** [comment]
I performed some tests and the values from the config file are working as intended.
If I use the following settings I get a max. of 10 entities for the normal Minecraft mobs.
```
#Spawn optimization
#Playing with difficulty HARD could exceed the max number of hostile entity from the 
#config file by 1.5x.
#These settings affects all mobs which are not covered by other parts of this config
[Spawn]
	#Defines the max. number of a mobs per entity type within player radius.
	#Range: 1 ~ 16
	maxEntityPerPlayer = 10
	#Defines the max. number of mobs per entity type within a world.
	#Range: 1 ~ 128
	maxEntityPerWorld = 40
	#General allow list for spawn entities (e.g. minecraft:squid) which should be ignored for optimization.
	allowList = [""]
	#General deny list for spawn entities (e.g. minecraft:bat) to no longer spawn in all worlds.
	denyList = [""]

	[Spawn.Minecraft]
		minecraftEnabled = true
		#e.g. mostly bats
		#Range: 1 ~ 16
		minecraftMaxPassiveMobsPerPlayer = 5
		#Range: 1 ~ 128
		minecraftMaxPassiveMobsPerWorld = 20
		#e.g. sheep, pig, horse, fox, ...
		#Range: 1 ~ 16
		minecraftMaxNeutralMobsPerPlayer = 10
		#Range: 1 ~ 128
		minecraftMaxNeutralMobsPerWorld = 40
		#e.g. slime, spider, zombie, ravager, ...
		#Range: 1 ~ 16
		minecraftMaxHostileMobsPerPlayer = 10
		#Range: 1 ~ 128
		minecraftMaxHostileMobsPerWorld = 40
		#e.g. mostly fish
		#Range: 1 ~ 16
		minecraftMaxWaterPassiveMobsPerPlayer = 5
		#Range: 1 ~ 128
		minecraftMaxWaterPassiveMobsPerWorld = 20
		#e.g. squid, dolphin, ...
		#Range: 1 ~ 16
		minecraftMaxWaterNeutralMobsPerPlayer = 5
		#Range: 1 ~ 128
		minecraftMaxWaterNeutralMobsPerWorld = 20
```

But I just released a new version which makes troubleshooting something like this much easier.

- You can use `/aptweaks entity overview` to see a list of currently spawned entities. In the server log you will also see the exact position if you want to double check them.
- With `/aptweaks spawnRules` you can see the max. number per Player for each enitity type, based on the config file.
- I also increased the max. allowed spawn numbers from the config file to allow higher limits

If you load the world and moving around for about 5 mins, you should see some mobs over `/aptweaks entity overview`.
Let me know if you see any unreasonable numbers to check this more in detail, but at the moment I'm unfortunately not able to reproduce the issue. :(

Please keep in mind that this mod consider all mobs for the calcuation. In the case you have a mobfarm it's expected that the mod is not allowing natural mobs over the limit near the mob farm.

**Comment by V972** [comment]
Hmm, I have the same problem, very little natural spawns for me.
Spawners are broken too, I ran through the dungeon from "Dungeon Crawl" w/o a single spawner activation.

Can it be that my laptop is _that_ bad the mod limits spaws this much?

**Comment by nanonestor** [comment]
Can confirm using in my own modpack with even the latest version of APT 0.4.4 and Forge 36.1.13 that practically no passive mobs spawn despite these all set as:

optimizePassiveMobs = false
optimizeNeutralMobs = false
minecraftEnabled = false

For whatever reason bees seem to be ignoring this extremely aggressive spawn banning.  (I do have the mod 'productive bees' present).  Nobody wants only 10 of each passive mob or such - setting the standard limit at 100 or 200 would be much more reasonable and still perform the function of protecting against world entity spam adequately imho.  And it would be extremely helpful to add on/off configuration for all of the things that APT tries to get it's fingers on. 

I really like some aspects of this mod however the entity side of what it does is extremely over aggressive and needs to allow more mobs max even when working properly - as using it on a multiplayer server where people want to farm mobs is almost impossible due to the max amounts allowed.  Having 100+ mobs of the same kind is not an unusual thing at all in regular modded minecraft, having 1000's not so much.

**Comment by MarkusBordihn** [comment]
@nanonestor 
Thanks a lot for the feedback and the very helpful insights. I agree that the limits are very aggressiv because my use-case is a little bit diffrent.
The servers I running are not having more than 8 players because of the large amount of mods (210+).
Most of the players are more using mods like tiny mob farm and not farming mobs in the traditional way.
But I totally understand your usecase and will try to find a solution to support these use cases as well.

**Comment by BugmanBugman** [comment]
Don't know if it's this mod, but for me the entire end seems to be empty

**Comment by MarkusBordihn** [comment]
Just to give an update on this topic. I  changed the logic and every entity which is not explicit defined will not be limited by the spawn control, so there is no default spawn limit any more.
This means if you disable the Minecraft rule, the natural spawn rate for all Minecraft entities is not touched at all, only the view distance will be in place.
Because of the other optimization the game has more resources and will spawn more mobs than usally, so this needs to be fine tuned if needed.
I also increased the max. number perPlayer and perWorld in the config to allow greater values.

![2021-05-14_18 44 07](https://user-images.githubusercontent.com/12183293/118308432-39cdc400-b4ec-11eb-87d9-d6c9d3f8559f.png)

If all tests are successfull the new version 0.4.x will be released tomorrow or so.

@BugmanBugman 
Could you please provide some more details in a seperate issue thread.
The mod has special spawn rules for the nether and the end and I'm unable to reproduce the issue, see:

![2021-05-14_19 35 04](https://user-images.githubusercontent.com/12183293/118308518-54a03880-b4ec-11eb-97b9-d4c4361f8836.png)

**Comment by BugmanBugman** [comment]
> Could you please provide some more details in a seperate issue thread.
> The mod has special spawn rules for the nether and the end and I'm unable to reproduce the issue, see:

The issue seems to have solved itself weirdly enough, it may have just me not staying long enough in the end. Mobs will begin to spawn in shortly after you spawning in.

**Comment by nanonestor** [comment]
> Just to give an update on this topic. I changed the logic and every entity which is not explicit defined will not be limited by the spawn control, so there is no default spawn limit any more.
> This means if you disable the Minecraft rule, the natural spawn rate for all Minecraft entities is not touched at all, only the view distance will be in place.
> Because of the other optimization the game has more resources and will spawn more mobs than usally, so this needs to be fine tuned if needed.
> I also increased the max. number perPlayer and perWorld in the config to allow greater values.
> 
> ![2021-05-14_18 44 07](https://user-images.githubusercontent.com/12183293/118308432-39cdc400-b4ec-11eb-87d9-d6c9d3f8559f.png)
> 
> If all tests are successfull the new version 0.4.x will be released tomorrow or so.
> 
> @BugmanBugman
> Could you please provide some more details in a seperate issue thread.
> The mod has special spawn rules for the nether and the end and I'm unable to reproduce the issue, see:
> 
> ![2021-05-14_19 35 04](https://user-images.githubusercontent.com/12183293/118308518-54a03880-b4ec-11eb-97b9-d4c4361f8836.png)

sounds like an improvement thanks!  any idea how far out the next update to include these changes will be?

**Comment by MarkusBordihn** [comment]
Unfortunately during the load test I found some issue, which needs to be fixed before I release the next version.
The curent plan is to release the next version in the next days, but the good news is that it's has a better performance (+4% - +10%) than the former versions.

![2021-05-20_17 44 01](https://user-images.githubusercontent.com/12183293/119009146-2068c400-b993-11eb-8ee7-6c73417b6929.png)

**Comment by MitztheKat** [comment]
Hi,
I have had this issue for a while **and** have had high success rates reproducing it via killing new players / entering the nether or end. both on a dedicated server.
If you could send a compiled copy to me for reproducing, that would be great.
thanks,
Mitz

**Comment by MarkusBordihn** [comment]
Just released the version 1.0.0, it should be live in the next hours:
https://www.curseforge.com/minecraft/mc-mods/adaptive-performance-tweaks/files/3324913

All of the spawn rates could be controlled within the config.
Unfortunately the server needs to be restarted after a config change.

I increased the default values and the max. numbers, futhermore spawns from spawners are ignored as long the server load is normal.

**Comment by MarkusBordihn** [comment]
I will close this for now. The upcoming version for 1.17.x will work differently which will hopefully cover more use cases without adjusting the whole configuration file.
